cmd/forwarder/config_test.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"testing"
+)
+
+func TestIssConfig_LogAuthUser_NoValidUser(t *testing.T) {
+	var base = IssConfig{
+		TokenUserSamplePct: 100,
+	}
+
+	var hits int
+	for i := 0; i < 100000; i++ {
+		if base.LogAuthUser("user") {
+			hits++
+		}
+	}
+
+	if hits > 0 {
+		t.Errorf("expected no hits, but got %d\n", hits)
+	}
+}
+
+func TestIssConfig_LogAuthUser_ValidUser(t *testing.T) {
+	var base = IssConfig{
+		ValidTokenUser:     "user",
+		TokenUserSamplePct: 100,
+	}
+
+	var hits int
+	for i := 0; i < 100000; i++ {
+		if base.LogAuthUser("user") {
+			hits++
+		}
+	}
+
+	if hits > 0 {
+		t.Errorf("expected no hits, but got %d\n", hits)
+	}
+}
+
+func TestIssConfig_LogAuthUser_ZeroPct(t *testing.T) {
+	var base = IssConfig{
+		ValidTokenUser:     "user",
+		TokenUserSamplePct: 0,
+	}
+
+	var hits int
+	for i := 0; i < 100000; i++ {
+		if base.LogAuthUser("not-user") {
+			hits++
+		}
+	}
+
+	if hits > 0 {
+		t.Errorf("expected no hits, but got %d\n", hits)
+	}
+}
+
+func TestIssConfig_LogAuthUser_OneHundredPct(t *testing.T) {
+	var base = IssConfig{
+		ValidTokenUser:     "user",
+		TokenUserSamplePct: 100,
+	}
+
+	var hits int
+	for i := 0; i < 100000; i++ {
+		if base.LogAuthUser("not-user") {
+			hits++
+		}
+	}
+
+	if hits != 100000 {
+		t.Errorf("expected 100000 hits, but got %d\n", hits)
+	}
+}
+
+func TestIssConfig_LogAuthUser_FiftyPct(t *testing.T) {
```

Comment:
@freeformz @sheax0r I'm concerned that this test might be too fragile.

Reply:
Actually don't do it that way. ;-) I'm making a diff.